source/frontend/src/components/maps/MapLeafletView.tsx

```diff
@@ -1,130 +1,141 @@
  */
 const MapLeafletView: React.FC<React.PropsWithChildren<MapLeafletViewProps>> = ({
-  lat,
-  lng,
-  zoom: zoomProp,
-  whenReady,
-  whenCreated,
-  geoFilter,
-  mapRef,
+  parentWidth,
 }) => {
-  const dispatch = useDispatch();
-
   const [baseLayers, setBaseLayers] = useState<BaseLayer[]>([]);
 
   const [activeBasemap, setActiveBasemap] = useState<BaseLayer | null>(null);
-  const smallScreen = useMediaQuery({ maxWidth: 1800 });
 
   const [bounds, setBounds] = useState<LatLngBounds>(defaultBounds);
-  const [layerPopup, setLayerPopup] = useState<LayerPopupInformation>();
 
   // a reference to the layer popup
   const popupRef = useRef<LeafletPopup>(null);
 
-  const history = useHistory();
-  const onPropertyViewClicked = (pid?: string | null, id?: number) => {
-    if (id !== undefined) {
-      history.push(`/mapview/sidebar/property/${id}?pid=${pid}`);
-    } else if (pid !== undefined && pid !== null) {
-      const parsedPid = pidParser(pid);
-      history.push(`/mapview/sidebar/non-inventory-property/${parsedPid}`);
-    } else {
-      console.warn('Invalid marker when trying to see property information');
-      toast.warn('A map parcel must have a PID in order to view detailed information');
-    }
-  };
+  const mapRef = useRef<LeafletMap | null>(null);
 
-  const { setState, selectedInventoryProperty, selectedFeature } = useContext(MapStateContext);
+  const [activeFeatureLayer, setActiveFeatureLayer] = useState<L.GeoJSON>();
 
-  if (mapRef.current && !selectedInventoryProperty) {
-    const center = mapRef.current.getCenter();
-    lat = center.lat;
-    lng = center.lng;
+  // add geojson layer to the map
+  if (!!mapRef.current && !activeFeatureLayer) {
+    setActiveFeatureLayer(geoJSON().addTo(mapRef.current));
   }
 
-  const parcelLayerFeature = selectedFeature;
-  const { showLocationDetails } = useActiveFeatureLayer({
-    selectedProperty: selectedInventoryProperty,
-    layerPopup,
-    mapRef,
-    parcelLayerFeature,
-    setLayerPopup,
-  });
-
-  const lastZoom = useAppSelector(state => state.mapViewZoom) ?? zoomProp;
-  const [zoom, setZoom] = useState(lastZoom);
+  const handleMapClickEvent = (latlng: LatLng) => {
+    mapMachine.mapClick(latlng);
+  };
+
+  const [zoom, setZoom] = useState(DEFAULT_MAP_ZOOM);
+  const [isMapReady, setIsMapReady] = useState(false);
+
+  const handleZoomUpdate = (zoomLevel: number) => {
+    setZoom(zoomLevel);
+  };
+
+  const mapMachine = useMapStateMachine();
+
+  const mapMachinePendingRefresh = mapMachine.pendingFitBounds;
+  const mapMachineProcessFitBounds = mapMachine.processFitBounds;
+  const mapMachineRequestedFitBounds = mapMachine.requestedFitBounds;
   useEffect(() => {
-    if (lastZoom === DEFAULT_MAP_ZOOM) {
-      dispatch(setMapViewZoom(smallScreen ? 4.9 : 5.5));
-    } else if (lastZoom !== zoom && zoom !== DEFAULT_MAP_ZOOM) {
-      dispatch(setMapViewZoom(zoom));
+    if (isMapReady && mapMachinePendingRefresh && mapRef.current !== null) {
```

Comment:
oh, ok. So we allow the fitbounds to fire but ignore it if the map isn't ready. hmm.

Reply:
right, ideally the ```mapRef.current !== null``` part would be reactive, but given it isnt...